patches/src/main/kotlin/app/revanced/patches/spotify/misc/lyrics/CustomLyricsPatch.kt

```diff
@@ -0,0 +1,133 @@
+package app.revanced.patches.spotify.misc.lyrics
+
+import app.revanced.patcher.extensions.InstructionExtensions.addInstructions
+import app.revanced.patcher.extensions.InstructionExtensions.getInstruction
+import app.revanced.patcher.extensions.InstructionExtensions.instructions
+import app.revanced.patcher.extensions.InstructionExtensions.replaceInstruction
+import app.revanced.patcher.fingerprint
+import app.revanced.patcher.patch.bytecodePatch
+import app.revanced.patcher.patch.stringOption
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod.Companion.toMutable
+import app.revanced.util.indexOfFirstInstructionReversedOrThrow
+import com.android.tools.smali.dexlib2.Opcode
+import com.android.tools.smali.dexlib2.builder.MutableMethodImplementation
+import com.android.tools.smali.dexlib2.builder.instruction.BuilderInstruction35c
+import com.android.tools.smali.dexlib2.iface.instruction.OneRegisterInstruction
+import com.android.tools.smali.dexlib2.immutable.ImmutableMethod
+import com.android.tools.smali.dexlib2.immutable.reference.ImmutableMethodReference
+import java.net.InetAddress
+import java.net.UnknownHostException
+import java.util.logging.Logger
+
+
+@Suppress("unused")
+val customLyricsPatch = bytecodePatch(
+    name = "Change lyrics provider",
+    description = "Changes the lyrics provider to a custom one.",
+) {
+    compatibleWith("com.spotify.music")
+
+    val lyricsUrlHost by stringOption(
+        key = "lyricsProviderUrl",
+        title = "Lyrics provider URL/hostname",
+        description = "The URL or the hostname to a custom lyrics provider.",
+        required = true,
+        default = "lyrics.natanchiodi.fr"
+    ) {
+        val host = it!!
+            .removePrefix("http://")
+            .removePrefix("https://")
+            .substringBefore("/")
+
+        try {
+            InetAddress.getByName(host)
+        } catch (e: UnknownHostException) {
+            Logger.getLogger(this::class.java.name).warning(
+                "Host $host did not resolve to any domain."
+            )
+        }
+        true
+    }
+
+    execute {
+        var patchedClientMethod: ImmutableMethod?
+
+        clientBuilderFingerprint.apply {
+            /**
+             * Copy the method definition of the HTTP client builder for a valid hostname.
+             */
+            patchedClientMethod = ImmutableMethod(
+                method.definingClass,
+                "getCustomLyricsProviderClient",
```

Comment:
Consistent with the name of the original method
```suggestion
                "getCustomLyricsProviderHttpClient",
```

Reply:
The original method name is `a` :D, anyway I have adjusted it with this.